### PR TITLE
Add valuesort, new control result mode and value normalizer

### PR DESCRIPTION
**Pull request by Omega359**
I've adding some functionality to support running the sqlite test suite against another database and this PR is the initial part of that work. This PR adds:

- valuesort from PR #123 
- a new control `resultmode` of either `valuewise` or `rowwise` (defaulting to the latter)
- a new value normalizer to allow for different value normalization for different files

The last point is a breaking API change so I took the liberty to rev the version #. If that isn't desired feel free to revert that portion of this PR.

With this PR I am able to run a slightly cleansed sqlite test suite using the Datafusion sqllogictest harness.

### CODE REVIEW

**Comment by Omega359** [comment]
@xxchan - any chance this could get a review?

**Comment by alamb** [comment]
For context, we are trying to run DataFusion against the entirety of the sqlite test corpus:
- https://github.com/apache/datafusion/issues/13812
-

File: tests/slt/rowsort.slt [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+
+control resultmode rowwise
+
 query III rowsort
 select * from example_sort
 ----
 1 10 2333
 10 100 2333
-2 20 2333
+2 20 2333
```
Comment (skyzh) [inline tests/slt/rowsort.slt:23]: could you please also help add a test case for `valuesort`? (might require some modifications for `FakeDB`)

**Review by skyzh — APPROVED** [review]
LGTM, thanks, and happy holiday :)

A general question: does sqlite only do string comparison? What's their behavior of `valuesort` on an integer column? Currently we only sort by string, and could order `10` before `2`. If they handle column types other than string, we probably need to add a custom comparator in the future?

**Comment by skyzh** [comment]
just released a new version :)

**Comment by alamb** [comment]
> just released a new version :)

Wow -- talk about speedy service with a smile. Thank you @skyzh  -- very much apprecaited

**Pull request merged**